core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -465,10 +465,15 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
       reconfigurable.reconfigure(newConfig)
   }
 
-  private def updatedConfigs(newProps: java.util.Map[String, _], currentProps: java.util.Map[_, _]): mutable.Map[String, _] = {
-    newProps.asScala.filter {
+  private def updatedConfigs(newProps: java.util.Map[String, _], currentProps: java.util.Map[String, _]):
+  (mutable.Map[String, _], Set[String]) = {
+    val deletedKeySet = currentProps.asScala.filter {
```

Comment:
This isn't quite deleted keys, right?  This is keys that are in the current map, but have a different value (or no value) in the new map.

Deleted keys would be something like `currentProps.asScala.filter(!newProps.contains(_))`

Reply:
Ah yes that's right, my bad. I've fixed this to reflect what `deletedKeys` should be.